vllm/core/block/cpu_gpu_block_allocator.py

```diff
@@ -147,6 +147,7 @@ def allocate_immutable_blocks(self, prev_block: Optional[Block],
             List[Block]: The newly allocated list of immutable blocks 
                 containing the provided block token IDs.
         """
+        assert device is not None
```

Comment:
From a quick search, it appears that this function is only called with `device` argument inside `BlockTable` class. Here, `device` is not `None`. So, it should be fine to change the type annotation of `device` from `Optional[Device]` to `Device`, removing the need for this assertion in the first place.